### ISSUE TITLE
[bug]RHS references in Antistasi Blufor Vanilla templates

### ISSUE THREAD

**Issue by gospelg** [official-antistasi-community/A3-Antistasi#450]
*Version:* 2.1.2

*Mods:* None

*Environment*: SP, MP LAN, MP host, MP dedi.

*.rpt attatched?* No

*have you edited the missionfile?*: No

Hey guys. The BLUFOR version fails to initialize on startup without RHS. I've found the reason to be lines 17-30 on the file /templates/OccupantsVanillaAltisB.sqf
```SQF
//NATO PvP Loadouts
NATOPlayerLoadouts = [
	//Team Leader
	["rhs_gref_teamLeader"] call A3A_fnc_getLoadout,
	//Medic
	["rhs_gref_medic"] call A3A_fnc_getLoadout,
	//Autorifleman
	["rhs_gref_breacher"] call A3A_fnc_getLoadout,
	//Marksman
	["rhs_gref_sniper"] call A3A_fnc_getLoadout,
	//Anti-tank Scout
	["rhs_gref_demolition"] call A3A_fnc_getLoadout,
	//Anti-tank Scout
	["rhs_gref_demolition"] call A3A_fnc_getLoadout
];
```
It fails as it cannot find the rhs_gref classnames. If you comment out these lines, the mission loads normally.

**Comment by Toakan** [comment]
@gospelg Can you grab the server RPT for this please?

**Comment by gospelg** [comment]
Sure, here it is. The error occurs at line 925. Let me specify that this was observed on singleplayer. I tried firing it up as a server locally on LAN and while the symptoms were not exactly the same, the mission still failed to initialize properly.

[Arma3_x64_2019-11-02_17-21-29.zip](https://github.com/official-antistasi-community/A3-Antistasi/files/3801000/Arma3_x64_2019-11-02_17-21-29.zip)

**Comment by Toakan** [comment]
Cheers, this is the bit we need.
```
17:23:00 Error in expression <g",1,1],
["DemoCharge_Remote_Mag",2,1],
]
+ _aceDefusalKit
+ _aceClacker
],

"H_>
17:23:00   Error position: <]
+ _aceDefusalKit
+ _aceClacker
],

"H_>
17:23:00   Error Missing [
17:23:00 File steam\__cur_sp.Altis\Templates\Loadouts\rhs_gref_demolition.sqf..., line 56
17:23:00 Error in expression <g",1,1],
["DemoCharge_Remote_Mag",2,1],
]
+ _aceDefusalKit
+ _aceClacker
],

"H_>
17:23:00   Error position: <]
+ _aceDefusalKit
+ _aceClacker
],

"H_>
17:23:00   Error Missing [
17:23:00 File steam\__cur_sp.Altis\Templates\Loadouts\rhs_gref_demolition.sqf..., line 56
17:23:00 Error in expression <ionNamespace setVariable [_loadoutName, _loadoutArray];
};

_loadoutArray;>
17:23:00   Error position: <_loadoutArray];
};

_loadoutArray;>
17:23:00   Error Undefined variable in expression: _loadoutarray
17:23:00 File Templates\Loadouts\getLoadout.sqf [A3A_fnc_getLoadout]..., line 95
17:23:00 Error in expression <able [_loadoutName, _loadoutArray];
};

_loadoutArray;>
17:23:00   Error position: <_loadoutArray;>
17:23:00   Error Undefined variable in expression: _loadoutarray
17:23:00 File Templates\Loadouts\getLoadout.sqf [A3A_fnc_getLoadout]..., line 98
17:23:00 Error in expression <rray isEqualTo []) then {
_loadoutArray = call compile preprocessFileLineNumbers>
17:23:00   Error position: <= call compile preprocessFileLineNumbers>
17:23:00   Error Generic error in expression
17:23:00 File Templates\Loadouts\getLoadout.sqf [A3A_fnc_getLoadout]..., line 94
```

**Comment by Toakan** [comment]
You're not using ACE? @gospelg

**Comment by gospelg** [comment]
No, I'm not using ace.

**Comment by SonorousMeerkat** [comment]
So the core of this issue, as correctly identified by gospelg, is that it is trying to call RHS classnames. This occurred when the template files had the new PvP loadout system implemented, it would seem that this one got missed when they were given classnames to work from. Should be a simple fix.

**Comment by gospelg** [comment]
Yeah, not sure what the deal with the ace stuff is, but this one is definitely template related. I saw in your fix you said it should take care of some of the friendly fire shenanigans. That was something else I was going to bring up. 

I made the changes that you had in your fix locally and I still had guys shooting each other upon recruiting them, but only with RHS enabled. Granted, the version I have is 2.1.2 which i grabbed off steam and unpacked to work with (I have my own fork of the mission), but this might need some more looking into. 

I haven't made a bug report yet because I haven't been able to pin it down yet, but heads up.
